### PR TITLE
NMS-16417: Make the refresh work again

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-16417

### CODE REVIEW

**Review by synqotik — CHANGES_REQUESTED** [review]
See comments - looks like this works, but I believe it could be updated to make the intent of the code more clear.

**Comment by christianpape** [comment]
> See comments - looks like this works, but I believe it could be updated to make the intent of the code more clear.

The function was pretty versatile, but for the given case it wasn't necessary. I simplified the code.

**Review by synqotik — APPROVED** [review]
Looks good - just one comment, not sure if it's needed or not. If you need me to re-approve, please ping me!

Also some failing tests, but they may not be related to this PR at all.

File: opennms-webapp/src/main/webapp/WEB-INF/jsp/alarm/detail.jsp [hunk 1]
```diff
@@ -135,6 +135,18 @@
...
 <jsp:directive.include file="/includes/bootstrap.jsp" />
 
+<script type="text/javascript">
+
+    function hasIdParam(url) {
+        return /.*[&?]id=.*/.test(url);
+    }
+
+    if (!hasIdParam(window.location.href)) {
+        window.location.href = document.location.href + '?id=${alarmId}';
```
[the remark this answers is not in the document]
Comment (mvrueden) [reply opennms-webapp/src/main/webapp/WEB-INF/jsp/alarm/detail.jsp:145]: In my opinion best practice would be to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams, which could handle that automatically.

File: opennms-webapp/src/main/webapp/WEB-INF/jsp/alarm/detail.jsp [hunk 2]
```diff
@@ -136,15 +136,11 @@
...
 
 <script type="text/javascript">
-
-    function hasIdParam(url) {
-        return /.*[&?]id=.*/.test(url);
-    }
-
-    if (!hasIdParam(window.location.href)) {
-        window.location.href = document.location.href + '?id=${alarmId}';
+    let url = new URL(location.href);
```
Comment (synqotik) [inline opennms-webapp/src/main/webapp/WEB-INF/jsp/alarm/detail.jsp:139]: Nice!!

**Review by synqotik — APPROVED** [review]
Looks good!

**Pull request merged**